clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -520,9 +535,15 @@ class MarkDeviceFunction : public RecursiveASTVisitor<MarkDeviceFunction> {
         }
       }
     } else if (const auto *FPTy = dyn_cast<FunctionProtoType>(Ty)) {
-      if (FPTy->isVariadic() && SemaRef.getLangOpts().SYCLIsDevice)
-	SemaRef.SYCLDiagIfDeviceCode(Loc.getBegin(), diag::err_sycl_restrict)
-            << Sema::KernelCallVariadicFunction;
+      if (FPTy->isVariadic() && SemaRef.getLangOpts().SYCLIsDevice) {
```

Comment:
I see that we check `SemaRef.getLangOpts().SYCLIsDevice` everywhere in this function, but I have no idea why we do it, because it seems that we can't get into this code without SYCLIsDevice lang option...